lightning-invoice/src/utils.rs

```diff
@@ -1886,4 +1891,154 @@ mod test {
 			_ => panic!(),
 		}
 	}
+
+    const TEST_RESULT_SIZE: usize = 3;
+
+    #[test]
+    fn test_zip_vecs_one() {
+        // test two nested vectors
+        let a = vec![vec!["a0", "b0", "c0"], vec!["a1", "b1"]];
+        let mut result = zip_vectors(a, TEST_RESULT_SIZE);
+
+        assert_eq!(Some("a0"), result.next());
+        assert_eq!(Some("a1"), result.next());
+        assert_eq!(Some("b0"), result.next());
+        assert_eq!(None, result.next());
```

Comment:
Instead of asserting against each iteration, simple `collect` the entire iterator in a vec and compare against a `vec![]` expression.